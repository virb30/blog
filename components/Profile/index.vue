<template>
    <div class="w-full flex items-center justify-center flex-col p-2">
        <img src="~/assets/images/profile.jpg" alt="avatar" class="h-40 w-40 rounded-full mb-8" />
        <h1 class="mb-4 text-3xl font-bold">
            Vinícius Bôscoa
        </h1>
        <span class="block font-thin mb-4 text-lg mt-2 text-center">
            Desenvolvimento Web, Segurança da Informação e Data Science
        </span>
        <div class="flex mt-6">
            <a href="https://www.linkedin.com/in/vinicius-boscoa"
                class="block bg-gray-300 rounded py-2 px-3 text-xl text-gray-800" @click="contactAccessed('linkedin')"
                id="contact-linkedin">
                <font-awesome-icon :icon="['fab', 'linkedin-in']" />
            </a>
            <a href="https://www.github.com/virb30" class="block bg-gray-300 rounded py-2 px-3 text-xl text-gray-800"
                @click="contactAccessed('github')" id="contact-github">
                <font-awesome-icon :icon="['fab', 'github']" />
            </a>
        </div>
    </div>
</template>

<script setup lang="ts">
import { useSessionStore } from '~/stores/session';

const sessionStore = useSessionStore();
const analytics = useAnalytics();

const contactAccessed = (type: string) => {
    const sessionId = sessionStore?.sessionId;

    if (sessionId) {
        analytics.logEvent('contact', {
            type,
            sessionId
        })
    }
}

</script>

<style scoped lang="postcss">
a+a {
    @apply ml-3;
}

a:hover {
    @apply bg-indigo-700 text-white;
}
</style>
