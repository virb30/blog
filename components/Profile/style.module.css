a + a {
	@apply ml-3;
}

a:hover {
	@apply bg-indigo-700 text-white;
}
