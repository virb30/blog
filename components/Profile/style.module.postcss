.container {
	@apply w-full flex items-center justify-center flex-col;

	& img {
		@apply h-40 w-40 rounded-full mb-8;
	}

	& h1 {
		@apply mb-4 text-5xl;
	}

	& span {
		@apply block font-thin mb-4 text-lg mt-2 text-center;
	}
}

.socialContainer {
	@apply flex mt-6;

	& a {
		@apply block bg-gray-200 rounded py-2 px-3 text-2xl text-gray-800;

		& + a {
			@apply ml-3;
		}

		&:hover {
			@apply bg-indigo-700 text-white;
		}
	}
}
