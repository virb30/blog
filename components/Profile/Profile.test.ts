import { mount } from "@vue/test-utils";
import { describe, expect, it, vitest } from "vitest";
import Profile from './index.vue';

describe('Profile Component', () => {
    it('should render childs', () => {
        const wrapper = mount(Profile, {
            props: {
                sessionStore: {
                    sessionId: '1'
                },
                analytics: {
                    logEvent: vitest.fn()
                }
            },
            global: {
                stubs: {
                    'font-awesome-icon': true,
                    sessionStorage
                },
            }
        });

        expect(wrapper.html).toMatchSnapshot();
    });

    it('should create component and log event childs', () => {
        const logEventSpy = vitest.fn()
        const wrapper = mount(Profile, {
            props: {
                sessionStore: {
                    sessionId: '1'
                },
                analytics: {
                    logEvent: logEventSpy
                }
            },
            global: {
                stubs: {
                    'font-awesome-icon': true,
                    sessionStorage
                },
            }
        });

        wrapper.get('#contact-linkedin').trigger('click');

        expect(logEventSpy).toBeCalledWith('contact', { type: 'linkedin', sessionId: '1' });

        expect(wrapper.html).toMatchSnapshot();
    });
});
