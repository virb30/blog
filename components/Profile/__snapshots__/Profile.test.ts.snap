// Vitest Snapshot v1

exports[`Profile Component > should create component and log event childs 1`] = `[Function]`;

exports[`Profile Component > should render childs 1`] = `[Function]`;
